adapter/protobuf/src/Mu/Adapter/ProtoBuf.hs

```diff
@@ -287,8 +291,13 @@ instance {-# OVERLAPS #-}
          => ProtoBridgeField sch ty ('FieldDef name ('TList t)) where
   fieldToProto (Field (FList xs))  = foldMap (oneFieldValueToProto fieldId) xs
     where fieldId = fromInteger $ natVal (Proxy @(FindProtoBufId sch ty name))
-  protoToField = Field . FList <$> (PBDec.repeated protoToOneFieldValue `at` fieldId <|> pure [])
+  protoToField = Field . FList <$> go
     where fieldId = fromInteger $ natVal (Proxy @(FindProtoBufId sch ty name))
+          base = PBDec.repeated protoToOneFieldValue `at` fieldId <|> pure []
+          go | supportsPacking (Proxy @(FieldValue sch t))
+             = PBDec.one protoToPackedFieldValue [] `at` fieldId <|> base
```

Comment:
So this will attempt to decode it as a packed field first, and fallback to decoding as unpacked if that fails?

I think that's fine, but if you have access to the protobuf wire type, you could also switch based on that. If it's `length_delimited`, it's a packed field. That's what we do in the Scala implementation: https://github.com/47degrees/pbdirect/blob/185c2f5cf76ecd81c6ce5269fb6e6ab49cc657de/src/main/scala/pbdirect/PBFieldReader.scala#L25-L26

Reply:
Looking at the code in [`runGetPacked`](http://hackage.haskell.org/package/proto3-wire-1.1.0/docs/src/Proto3.Wire.Decode.html#runGetPacked), that's sort of what's happening. The `runGetPacked` function checks whether we are in the "length delimited" case, and if not it fails. This failure is then taken over by the `(<|>)`, which tries the other option.